graphics/tinygl/ztriangle.cpp

```diff
@@ -34,14 +34,31 @@ namespace TinyGL {
 
 static const int NB_INTERP = 8;
 
-template <bool kDepthWrite, bool kSmoothMode, bool kFogMode, bool kEnableAlphaTest, bool kEnableScissor, bool kEnableBlending, bool kStencilEnabled, bool kDepthTestEnabled>
+static bool applyStipplePattern(int x, int y, void *stipplePattern) {
+
+	byte *stipple = (byte *)stipplePattern;
+	int stippleX = x % 32;
+	int stippleY = y % 32;
+	int byteIndex = stippleY * 4 + (stippleX / 8); // 4 bytes per row
+	int bitIndex = stippleX % 8;
+
+    byte bitmask = 1 << (7 - bitIndex);
+	return (stipple[byteIndex] & bitmask);
```

Comment:
I believe C++ handles this as an integral conversion instead of a logical test, which makes this undefined behavior of the result of the bitmask is something other than `(int)true`.  I think you need to do `(stipple[byteIndex] & bitmask) != 0`.

Reply:
Yeah I confirmed this is a false positive, `bool` has special treatment.  I think `!= 0` would be clearer, but it will get logic-tested, so it's not broken.